usefulUtilities/statusIndicator/webApp/app/statusIndicator_webApp.js

```diff
@@ -0,0 +1,325 @@
+        ORDER BY teamName, displayName`;
+
+    connection.query(query, function(error, results, fields) {
+        if (error) {
+            var responseObject = {
+                status: "error",
+                text: "Error while retrieving ALL employees! " + JSON.stringify(error)
+            };
+
+            response.statusCode = 500;
+            response.setHeader('Content-Type', 'application/json');
+            return response.end(JSON.stringify(responseObject));
+        }
+
+        if (results.length === 0) {
+            return;
+        }
+
+        var teamName = results[0].teamName;
+        var responseHTML = `<div class="team"><h2>${teamName}</h2><table>`;
+        for (var i = 0; i < results.length; i++) {
+            if (teamName !== results[i].teamName) {
+                responseHTML += `</table></div><div class="team"><h2>${results[i].teamName}</h2><table>`
+                teamName = results[i].teamName;
+            }
+            responseHTML += `
+<tr>
+    <td width="60%">${results[i].displayName}</td>
+    <td width="40%">${results[i].status}</td>
+</tr>
+            `;
+        }
+        responseHTML += `</table></div>`
+
+        response.statusCode = 200;
+        response.setHeader('Content-Type', 'text/html');
+        return response.end(responseHTML);
+    });
+}
+
+
+// Get employees from team
+function getTeamEmployees(teamName, response) {
+    var query = `SELECT * FROM statusIndicator
+        WHERE teamName = '${teamName}' ORDER BY displayName`;
+
+    connection.query(query, function(error, results, fields) {
+        if (error) {
+            var responseObject = {
+                status: "error",
+                text: "Error while retrieving employees with teamName! " + JSON.stringify(error)
+            };
+
+            response.statusCode = 500;
+            response.setHeader('Content-Type', 'application/json');
+            return response.end(JSON.stringify(responseObject));
+        }
+
+        var responseHTML = "<table><tr><th>Display Name</th><th>status</th></tr>";
+        for (var i = 0; i < results.length; i++) {
+            responseHTML += `
+<tr>
+    <td>${results[i].displayName}</td>
+    <td>${results[i].status}</td>
+</tr>
+            `;
+        }
+        
+        responseHTML += "</table>";
+
+        response.statusCode = 200;
+        response.setHeader('Content-Type', 'text/html');
+        return response.end(responseHTML);
+    });
+}
+
+
+// Handles any GET requests made to the server endpoint
+// The handled method types are:
+// "heartbeat"
+// "getAllEmployees"
+// "getTeamEmployees"
+function handleGetRequest(request, response) {
+    var queryParamObject = url.parse(request.url, true).query;
+    var type = queryParamObject.type;
+
+    // root/type=?alsjdf
+    switch(type) {
+        case "heartbeat":
+            heartbeat(queryParamObject, response);
+        break;
+
+        case "getAllEmployees": // http://localhost:3305/?type=getAllEmployees
```

Comment:
the URL in this comment will be slightly different when we deploy - we'll talk more about this later